storage/src/adb/any/sync/mod.rs

```diff
@@ -34,6 +54,15 @@ pub enum Error {
     /// Invalid client state
     #[error("Invalid client state")]
     InvalidState,
+    /// Sync target hash unchanged
+    #[error("Sync target hash unchanged")]
+    SyncTargetHashUnchanged,
+    /// Sync target moved backward
+    #[error("Sync target moved backward: {old:?} -> {new:?}")]
+    SyncTargetMovedBackward {
+        old: Box<dyn fmt::Debug + Send + Sync>,
```

Comment:
With the second instance of this, I'm tempted to say we should make `Error` over `Digest`?